[hash-library] build failure

**Host Environment**
OS: MacOS (Darwin-20.4.0)
Compiler: clang 12.0.5.12050022
CMake: 3.22.3

**To Reproduce**
Steps to reproduce the behavior:
Clone the vcpkg repo master branch.
Install the hash-library package: `./vcpkg install hash-library`

In the **CMakeList.txt** add the hash-library dependency:
```
find_package(unofficial-hash-library CONFIG REQUIRED)
target_link_libraries(app PRIVATE unofficial::hash-library)
```
**Failure logs**
```
CMake Error at /Users/user/git/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
  Could not find a package configuration file provided by
  "unofficial-hash-library" with any of the following names:

    unofficial-hash-libraryConfig.cmake
    unofficial-hash-library-config.cmake

  Add the installation prefix of "unofficial-hash-library" to
  CMAKE_PREFIX_PATH or set "unofficial-hash-library_DIR" to a directory
  containing one of the above files.  If "unofficial-hash-library" provides a
  separate development package or SDK, be sure it has been installed.
```

**Additional context**
Besides this library I can use other libraries from vcpkg just fine:
```
find_package(cpr CONFIG REQUIRED)
find_package(nlohmann_json CONFIG REQUIRED)
```

If I try another package name like `find_package(hash-library CONFIG REQUIRED)` it won't help 😞

--- a/ports/hash-library/CMakeLists.txt
+++ b/ports/hash-library/CMakeLists.txt
@@ -43,3 +43,6 @@ install(
     NAMESPACE unofficial::
     DESTINATION share/unofficial-hash-library
 )
+
+file(WRITE "unofficial-hash-library-config.cmake" "include(\${CMAKE_CURRENT_LIST_DIR}/unofficial-hash-library-targets.cmake)\n")
+install(FILES "unofficial-hash-library-config.cmake" DESTINATION share/unofficial-hash-library)

--- a/ports/hash-library/CMakeLists.txt
+++ b/ports/hash-library/CMakeLists.txt
@@ -44,5 +44,5 @@ install(
     DESTINATION share/unofficial-hash-library
 )
 
-file(WRITE "unofficial-hash-library-config.cmake" "include(\${CMAKE_CURRENT_LIST_DIR}/unofficial-hash-library-targets.cmake)\n")
-install(FILES "unofficial-hash-library-config.cmake" DESTINATION share/unofficial-hash-library)
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/unofficial-hash-library-config.cmake" "include(\${CMAKE_CURRENT_LIST_DIR}/unofficial-hash-library-targets.cmake)\n")
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/unofficial-hash-library-config.cmake" DESTINATION share/unofficial-hash-library)

--- a/ports/hash-library/vcpkg.json
+++ b/ports/hash-library/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "hash-library",
   "version": "8",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Portable C++ hashing library",
   "homepage": "https://create.stephan-brumme.com/hash-library/",
   "license": "Zlib",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2790,7 +2790,7 @@
     },
     "hash-library": {
       "baseline": "8",
-      "port-version": 1
+      "port-version": 2
     },
     "hayai": {
       "baseline": "2019-08-10",

--- a/versions/h-/hash-library.json
+++ b/versions/h-/hash-library.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "909bfeef471a1b379f491db05c4edaef1595d314",
+      "git-tree": "b3e49529dd25b60564de8ceed9485e17e8a7f8da",
       "version": "8",
       "port-version": 2
     },

--- a/versions/h-/hash-library.json
+++ b/versions/h-/hash-library.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "909bfeef471a1b379f491db05c4edaef1595d314",
+      "version": "8",
+      "port-version": 2
+    },
+    {
       "git-tree": "fad55dc152114a3f71c5eafd33a3facd7e3a2286",
       "version": "8",
       "port-version": 1

